modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -3318,12 +3348,14 @@ def _compute_new_widths():
             new_partitions, new_index, new_columns, new_lengths, new_widths, new_dtypes
         )
 
+    @lazy_metadata_decorator(apply_axis="both")
```

Comment:
Was this not necessary before, or did we simply not have tests for this case?

Reply:
this method was not implemented before (and so wasn't used at all), the decorator was added accordingly to the needs of this method in terms of the actual metadata (groupby has to know the latest indices and columns)
https://github.com/modin-project/modin/blob/ab90115e5493b36a173539cfa2bbc68086dbfb52/modin/core/dataframe/pandas/dataframe/dataframe.py#L3361